.github/workflows/security.yml

```diff
@@ -0,0 +1,44 @@
+name: Security
+on:
+  push:
+    tags-ignore:
+      - "**"
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+permissions:
+  contents: read
+
+jobs:
+  security-scan:
+    name: security-scan
+    strategy:
+      matrix:
+        go: ['1.20']
+        os: [ubuntu-latest]
```

Comment:
I have added this `matrix` to all of the workflows so that we can easily add additional `go versions` or `operating systems` in the future. As far as I know having this section with only one go version and one os does not adversely affect anything. If anyone else knows differently please let me know.

Reply:
But `- uses: ./.github/actions/setup` won't pickup the Go version from the matrix. Not sure if we really need that additional complexity.

Reply:
I have updated the .github/actions/setup to accept a `go-version`